component/file_cache/file_cache.go

```diff
@@ -442,8 +442,32 @@ func (fc *FileCache) DeleteDir(options internal.DeleteDirOptions) error {
 
 // StreamDir : Add local files to the list retrieved from storage container
 func (fc *FileCache) StreamDir(options internal.StreamDirOptions) ([]*internal.ObjAttr, string, error) {
+	// For stream directory, there are three different child path situations we have to potentially handle.
+	// 1. Path in storage but not in local cache
+	// 2. Path not in storage but in local cache (this could happen if we recently created the file [and are currently writing to it]) (also supports immutable containers)
+	// 3. Path in storage and in local cache (this could result in dirty properties on the service if we recently wrote to the file)
+
+	// To cover case 1, grab all entries from storage
 	attrs, token, err := fc.NextComponent().StreamDir(options)
 
+	for _, attr := range attrs {
+		entryPath := common.JoinUnixFilepath(attr.Path)
```

Comment:
Do we need to normalize the path here? I think the name here should already be normalized.

Reply:
Good catch.